src/components/_classes/component/Component.js

```diff
@@ -2020,7 +2020,7 @@ export default class Component extends Element {
    */
   updateComponentValue(value, flags) {
     flags = flags || {};
-    let newValue = (value === undefined || value === null) ? this.getValue() : value;
+    let newValue = ((value === undefined || value === null) && this.component.type !== 'number') ? this.getValue() : value;
```

Comment:
This should just be the NumberComponent overriding ```normalizeValue``` and not code within the Component that is specific to component.type.

Reply:
@travist Done.